[DNM] scripts: Add scripts and workflows used for results verification

PerMac opened the pull request:
The scripts were ported from test_results repo with minor adjustments to match json format of results

PerMac committed:
scripts: Add scripts and workflows used for results verification

The scripts were ported from test_results repo with minor adjustments
to match json format of results.

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

PerMac commented:
I didn't port the /doc/ folder, it seems irrelevant and not used.
I removed /doc/ from the CODOWNERS section and updated list of codeowners (Maksim no longer works on Zephyr)
I fixed the warning: ```Warning: The `set-output` command is deprecated and will be disabled soon.``` in
```check_results.yml ``` file.
@nashif Does cache step make sense? ~~I always see it reports ``` Cache not found for input keys: Linux-test-pip ```, also in the original repo. I tried reading about it https://github.com/orgs/community/discussions/27059 and it seems it has a "peculiar" functionality, and most commenters found it useless.~~ edit: I saw it worked on the latest run on my fork. Is the point of it to only work on reruns on PRs?

PerMac commented:
I verified the scripts in my fork of this repo https://github.com/PerMac/hardware_test_results/pull/2. I merged this PR there and opened another adding json results

PerMac commented:
@nashif I have no rights in this repo and cannot add reviewers nor I am recognized in the CODOWNERS file. Can you add me? I guess the same applies to @hakehuang

PerMac commented:
I also updated /cache /checkout and /file-changes-action to the latest ones

hakehuang commented on scripts/results_verification.py:
```diff
@@ -0,0 +1,141 @@
+    err_cnt = 0
+    all_clear = False
+    for ts in report["testsuites"]:
+        if ts["platform"] != platform_name:
```
platform name need normized for HWv2

PerMac replied:
addressed

PerMac commented:
This PR won't show the workflow in action on examplary results. A workflow has to be first merged for github to detect it and execute. I merged this PR on my fork and raised a PR with exemplary results to verify the workflow. The results can be seen here https://github.com/PerMac/hardware_test_results/actions/runs/9892293815/job/27324614420?pr=5

hakehuang reviewed (approved):
looks good to me. can wee add a document as well.

arikgreen commented on .github/workflows/check_results.yml:
```diff
@@ -0,0 +1,63 @@
+        pip3 install junit2html
+
+    - id: file_changes
+      uses: trilom/file-changes-action@v1.2.4
```
@PerMac please change the action you are using to something else like  https://github.com/marketplace/actions/changed-files
because repository https://github.com/trilom/file-changes-action seams to be abandoned.
![image](https://github.com/user-attachments/assets/9d3eeb08-129b-4e05-b9aa-2c8fa42dc4fb)

PerMac replied:
done, thanks

PerMac committed:
verification: Add auto-merge step

The step enables auto-merge on a PR if the following conditions
are fulfilled:
- author of the PR is among trusted authors (defined in this workflow)
- 'results' label is present (workflow adds the label itself)

This commit also makes the action to give a positive review to the PR
if the uploaded files pass verification.
Auto-marge must be enable both on the repo and organization settings
level. Auto-merge can only happen if all the branch protection rules
are fulfilled.
(required checks passed, amount of positive reviews, etc.)

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

PerMac commented:
I added the step enabling auto-merge if required conditions are fulfilled (described in commit's msg). I put this PR to draft since I want to test this more on my fork repo.

arikgreen reviewed (approved):
looks good for me
